File: streamx-spark/streamx-spark-monitor/streamx-spark-monitor-scala/src/main/scala/com/streamxhub/spark/monitor/core/domain/SparkMonitor.scala
package com.streamxhub.spark.monitor.core.domain

import java.io.Serializable
import java.util.Date

import javax.validation.constraints.NotBlank
import lombok.Data

@Data
class SparkMonitor(
                    @NotBlank(message = "{required}") val monitorId: Int,
                    @NotBlank(message = "{required}") val appId: String,
                    @NotBlank(message = "{required}") val appName: String,
                    @NotBlank(message = "{required}") val confVersion: String
                  ) extends Serializable {
  var status: Int = 0
  var createTime: Date = null
  var modifyTime: Date = null

  def this() {
    this(0, null, null, null)
  }

}

File: streamx-spark/streamx-spark-monitor/streamx-spark-monitor-core/src/main/java/com/streamxhub/spark/monitor/core/domain/SparkMonitor.java
package com.streamxhub.spark.monitor.core.domain;

import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import javax.validation.constraints.NotBlank;
import java.util.Date;

@Data
@TableName("t_spark_monitor")
public class SparkMonitor {

    @NotBlank(message = "{required}")
    private String monitorId;

    @NotBlank(message = "{required}")
    private String appId;

    @NotBlank(message = "{required}")
    private String appName;

    @NotBlank(message = "{required}")
    private String confVersion;

    private Integer status;

    private Date createTime;
    private Date modifyTime;

}
